src/IoC/IoC.java

```diff
@@ -0,0 +1,19 @@
+package IoC;
+
+import java.util.Map;
+
+public class IoC {
+    private static IoC instance;
+    private Map<String,Object> myMap;
+
+    public static IoC getInstance() {
+        if (instance == null) {
```

Comment:
Use `Optional` instead

Reply:
As the return value of getInstance()? Why? it will never be or return null, because if it checks that it doesnt exists it creates the instance

Reply:
Optional does not work only for nulls but works for optional values. Here there is a clear use case of its usage, if instance is null you return a value otherwise you return another value.

Please make usage of `Optionals`.

Also I recommend to take sometime and read about it since optionals aree widely used nowdays

Reply:
I read about optionals, and how they're used to see if an object exists or is present or not, and how convenient they are to manage null values also. But what i did is made the Ioc instance Optional, and the return value of getInstance() is still Ioc, but the existing or present value of the Optional

Reply:
If you don't send any args in the ctor you could do it even simpler:
```
public static IoC getInstance() {
  return Optional.ofNullable(this.instance).orElse(Ioc::new);
}
```

Reply:
Solved, but instead of writing "(Ioc::new)" i wrote "(new Ioc(instancesMap))", because the following error appeared: "Ioc is not a functional interface", because Ioc is a class not a functional interface